packages/js-runtime/typescript/compiler.ts

```diff
@@ -300,8 +303,36 @@ export class TypeScriptCompiler implements Compiler<TypeScriptCompilerOptions> {
       throw new Error("Missing main source.");
     }
 
+    // Check if the main source file has the /// <cts-enable /> directive
+    const hasCtsEnableDirective = (sourceFile: SourceFile): boolean => {
+      const text = sourceFile.getFullText();
+      const tripleSlashDirectives = ts.getLeadingCommentRanges(text, 0) || [];
+
+      for (const comment of tripleSlashDirectives) {
+        const commentText = text.substring(comment.pos, comment.end);
+        if (/^\/\/\/\s*<cts-enable\s*\/>/m.test(commentText)) {
+          return true;
+        }
+      }
+      return false;
+    };
+
+    // Only apply transformers if the file has the directive
+    const transformers = hasCtsEnableDirective(mainSource)
```

Comment:
This means a file can be invalid based on what's importing it -- i'd expect the declaration to be per-file, like other similar JS/TS declarations